x-pack/platform/plugins/shared/onechat/public/services/agents/profiles/agents_profiles_service.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { HttpSetup } from '@kbn/core-http-browser';
+import type { AgentProfile } from '@kbn/onechat-common';
+import type {
+  AgentProfileCreateRequest,
+  AgentProfileUpdateRequest,
+  AgentProfileListOptions,
+} from '../../../../common/agent_profiles';
+
+export class AgentProfilesService {
+  private readonly http: HttpSetup;
+
+  constructor({ http }: { http: HttpSetup }) {
+    this.http = http;
+  }
+
+  /**
+   * List all agent profiles
+   */
+  async list(options?: AgentProfileListOptions): Promise<AgentProfile[]> {
```

Comment:
NIT: I know those `XXXResponse` types I use in `shared/onechat/common/http_api` sounds like a chore, but that's the only effective way to make sure we "stay sync" between the browser-side API and the server-side route (by enforcing type on both side, e.g.)

https://github.com/elastic/kibana/blob/48e4ede08addca21dd699a023583ef532b642af1/x-pack/platform/plugins/shared/onechat/server/routes/agents.ts#L44-L46

and then

https://github.com/elastic/kibana/blob/2c02bce5f41dfb6bcf9b3058205c1e992be5e0a7/x-pack/platform/plugins/shared/onechat/public/services/agents/agents_service.ts#L25-L27

(so that changing a signature on one side causes a TS error on the other side)